packager/packager.cc

```diff
@@ -589,9 +591,17 @@ struct PackageInfoFinder {
     }
 };
 
+// TODO (aadi-stripe) we can avoid syscalls if we invent an efficient way of looking up
+// directories in the source tree via GlobalState. Might be tied to https://github.com/sorbet/sorbet/issues/4509
+bool pathExists(const std::string &path) {
+    struct stat buffer;
+    return (stat(path.c_str(), &buffer) == 0);
```

Comment:
Do you want to additionally check `S_ISDIR(buffer.st_mode)`?

Reply:
I think it's ok here because the only usage of this method passes paths ending in `/`, so it's unlikely that those are files and not directories.